resource/js/vocab-search.js

```diff
@@ -7,18 +7,119 @@ const vocabSearch = Vue.createApp({
+    document.addEventListener('click', this.onClickOutside)
+  },
+  beforeUnmount () {
+    document.removeEventListener('click', this.onClickOutside)
   },
   methods: {
     autoComplete () {
+      const delayMs = 300
+
+      // when new autocomplete is fired, empty the previous result
+      this.hideDropdown()
+
+      // cancel pending API calls when method is called
+      clearTimeout(this._timerId)
+
+      // is the search term is in cache, use the cache
+
+      // delay call, but don't execute if the search term is not at least two characters
+      if (this.searchTerm.length > 1) {
+        this._timerId = setTimeout(() => { this.search() }, delayMs)
+      }
+    },
+    /*
+     * search should fetch the response and save it to cache
+     * search calls renderResults for displaying the response
+     */
+    search () {
+      const mySearchCounter = this.searchCounter + 1 // make sure we can identify this search later in case of several ongoing searches
+      this.searchCounter = mySearchCounter
+
+      fetch('rest/v1/' + SKOSMOS.vocab + '/search?query=' + this.formatSearchTerm() + '&lang=' + SKOSMOS.lang)
+        .then(data => data.json())
+        .then(data => {
+          if (mySearchCounter === this.searchCounter) {
+            this.renderedResultsList = data.results // update results (update cache if it is implemented)
+            this.renderResults() // render after the fetch has finished
+          }
+        })
+    },
+    formatSearchTerm () {
+      if (this.searchTerm.includes('*')) { return this.searchTerm }
+      const formatted = this.searchTerm + '*'
+      return formatted
+    },
+    renderMatchingPart (searchTerm, label) {
+      const regex = new RegExp(searchTerm, 'i') // case insensitive matching
+      if (label && regex.test(searchTerm)) {
+        return label.replace(regex, (match) => `<b>${match}</b>`)
+      }
+      return label
+    },
+    translateType (type) {
+      return SKOSMOS.msgs[SKOSMOS.lang][type]
+    },
+    /*
+     * renderResults is used when the search string has been indexed in the cache
+     * it also shows the autocomplete results list
+     * TODO: Showing labels in other languages, extra concept information and such goes here
+     */
+    renderResults () {
+      // get the results list form cache if it is implemented
+      console.log(this.renderedResultsList)
+      const renderedSearchTerm = this.searchTerm // save the search term in case it changes while rendering
+      this.renderedResultsList.forEach(result => {
+        const hitPref = this.renderMatchingPart(renderedSearchTerm, result.prefLabel)
+        const hitAlt = this.renderMatchingPart(renderedSearchTerm, result.altLabel)
+        const hitHidden = this.renderMatchingPart(renderedSearchTerm, result.hiddenLabel)
+        if ('uri' in result) { // change uris to Skosmos page urls
+          result.uri = SKOSMOS.vocab + '/' + SKOSMOS.lang + '/page?uri=' + encodeURIComponent(result.uri)
```

Comment:
This is maybe a nitpick, but I think it would be cleaner to assign this to `const result_url`. It's better to avoid mutating already existing variables and data structures as it makes code logic easier to inspect.